### PR TITLE
feat(cli): implement add flag and config commands

### PR DESCRIPTION
**Pull request by CakeWithDivinity**
See #80 

The commands look like this:
image

### CODE REVIEW

File: packages/cli/src/add-flag.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  console.log(chalk.blue("Updating local config..."));
+  const updatedFileString = updateConfigFile(newConfig, configFileContents);
+  const formattedFile = await prettier.format(updatedFileString, {
+    filepath: configFilePath,
+  });
+  await fs.writeFile(configFilePath, formattedFile);
+  console.log(chalk.green("Local config updated successfully"));
+
+  console.log(chalk.blue("Updating remote config..."));
+  await push({ apiKey: options.apiKey, configPath: options.configPath, apiUrl: options.host });
```
Comment (cstrnt) [inline packages/cli/src/add-flag.ts:38]: I think there should be some sort of rollback here if the push fails

**Review by cstrnt — CHANGES_REQUESTED** [review]
I think this might be related to the `mergeConfigs` function but it resolves the .env vars and then alters the config. Maybe just do a atomic update on the config?

Screenshot 2023-10-10 at 11 04 44

**Comment by cstrnt** [comment]
Screenshot 2023-10-16 at 09 15 41
Something is still odd with the formatting after the file gets updated @CakeWithDivinity

**Comment by CakeWithDivinity** [comment]
@cstrnt Should be fixed now. We now only replace newlines in the `defineConfig(...)` part. The issue was that when the newlines were missing in the first line, it was still a correctly formatted file for prettier. This way, this should only happen in the part of the file that we actually want to edit.

**Review by cstrnt — APPROVED** [review]

**Comment by cstrnt** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #81** <a href="https://app.graphite.dev/github/pr/tryabby/abby/81?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/tryabby/abby/81?utm_source=stack-comment).

**Pull request merged**